private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/maven/MavenResolver.java

```diff
@@ -402,12 +402,61 @@ private List<Dependency> createManagedDependencies(
         .filter(dep -> !artifactKeys.contains(getArtifactKey(dep.getArtifact())))
         .forEach(managedDependencies::add);
 
-    return managedDependencies.build();
+    // Last, normalize versions across different classifiers of an artifact
```

Comment:
It feels better to pick the most recent version, and to output a `LogEvent` when we find more than one version of something

Reply:
I'm not sure, we don't pick the most recent version of a dependency in any other case, would it be confusing if we only picked the most recent version for artifacts with classifiers?   Should we just pick the most recent for all artifacts in the dependency list?

Reply:
We would also have to add logic to only update to the latest version if a version for one of the classified dependencies wasn't an explicit dependency listed by the user.  That feels more complicated than the feature is worth.

Reply:
Let's land what we have and iterate.